addons/web/static/src/core/assets.js

```diff
@@ -108,23 +108,45 @@ export const loadBundleTemplates = memoize(async function loadBundleTemplates(na
     return processTemplates(templates);
 });
 
+
+export const _loadBundle= memoize(async function(bundleName){
+    const bundleURL = new URL(`/web/bundle/${bundleName}/jscss`, window.location.origin);
+    const bundleInfo = await (await browser.fetch(bundleURL.href)).json();
+    /*[{"type": "script", "src": "/web/assets/266-d34b0b4/documents_spreadsheet.o_spreadsheet.min.js", "mimetype": "text/javascript"}]*/
+    return bundleInfo;
+});
+
 const bundlesCache = {};
+
+
 /**
  * Loads a bundle.
  *
  * @param {string} name the name of the bundle to load
  * @param {Bundle} options parts of the bundle to load (see Bundle typedef)
  * @returns {Promise<LoadedBundle>}
  */
-async function loadBundle(name, options) {
+export async function loadBundle(name, options) {
     const { templates } = options;
     if (!bundlesCache[name]) {
         bundlesCache[name] = { name };
     }
     if (templates && !bundlesCache[name].templates) {
         bundlesCache[name].templates = loadBundleTemplates(name);
     }
-    // TODO: if ("js/css") {...} to support lazy loading js/css from bundles
+
+    if (options.css || options.js) {
+        const bundleInfo = await _loadBundle(name);
+        const scriptInfo = bundleInfo.find((i)=> i.type ==="script")
```

Comment:
bundles may contain more than one script or style node (eg if a bundle has links to external code)

Reply:
Thanks for the review :) Do you have an example of a definition of a bundle that would have multiple outputs ?

Reply:
The only one that exists currently is `web.assets_frontend` in payment_adyen, there are a few test bundles for it too: `test_assetsbundle.bundle4` in test_assetbundle and `website.test_bundle` in website.